app/utility/api/serializer.rb

```diff
@@ -0,0 +1,45 @@
+module Api
+  class Serializer
+    attr_reader :attributes
+    MODEL_INFO = YAML.load(File.new(Rails.root.join('config', 'api.yml')).read).freeze
+
+    def initialize(model, exclude: [])
+      @model = model
+      @exclude = Array.wrap(exclude).map(&:to_s)
+      @class_name = @model.class.name.downcase
+      @fields = MODEL_INFO.dig(@class_name, 'fields')&.reject { |f| @exclude.include?(f) }
+      @ignore = attributes_to_ignore
+      set_attributes
+    end
+
+    private
+
+    def set_attributes
+      @attributes = @model.attributes.delete_if { |k, _| @ignore.include?(k) }
+      model_specific_attributes unless @fields.nil?
+      isoify_updated_at
+    end
+
+    def model_specific_attributes
+      @fields.each do |(field_name, code)|
+        @attributes.store(field_name, @model.instance_eval("self.#{code}"))
```

Comment:
ooh-lala! so meta!

Reply:
haha! oh, you know, just writing code in a yaml file and then eval'ing it directly on the model....nothing to see here...